docs/storage.md

```diff
@@ -157,6 +157,19 @@ promtool tsdb create-blocks-from openmetrics <input file> [<output directory>]
 
 After the creation of the blocks, move it to the data directory of Prometheus. If there is an overlap with the existing blocks in Prometheus, the flag `--storage.tsdb.allow-overlapping-blocks` needs to be set. Note that any backfilled data is subject to the retention configured for your Prometheus server (by time or size).
 
+#### Custom Block Durations
+
+The `promtool tsdb create-blocks-from` command allows for the number of blocks created when backfilling to be configured. By default, the tool will use the default block duration (2h) for the TSDB; this behavior is the most generally applicable and correct. However, in specific cases it may be advantageous to use a larger value for the block duration - when backfilling data over a long range of times, larger blocks will be much more efficient to operate over. When choosing a larger block duration, a number of constraints must be taken into consideration:
+
+- the block duration *must* be a power of two number of hours
+- blocks, when inserted must be aligned to the TSDB
+
+The --max-block-duration flag allows the user to configure some maximum duration of blocks. The backfilling tool will pick a block duration no larger than this that follows the above rules.
+
+While larger blocks may improve the performance of backfilling large datasets, drawbacks exist as well, as many algorithms in the TSDB will operate over the entire block as one monolithic entity. The larger a block is when written to the TSDB, the higher chances are that adverse effects will occur: time-based retention policies must keep the entire block around if even one part of the (potentially large) block is still valid. Conversely, size-based retention policies will remove the entire contents of the block even if the TSDB only goes over the size limit in a minor way.
+
+Therefore, backfilling with few blocks, thereby choosing a larger block duration, must be done with care and is not recommended for any production instances.
```

Comment:
Some suggested edits to make it more compact and removed some explanation about block range selection which is not relevant for this section of the doc, especially when it's `max-block-duration` now.

Regarding `"not recommended for any production instances"`, I am not sure, any comments on that @roidelapluie?

```suggestion
#### Longer Block Durations

By default, the promtool will use the default block duration (2h) for the blocks; this behavior is the most generally applicable and correct. However, when backfilling data over a long range of times, it may be advantageous to use a larger value for the block duration to backfill faster and prevent additional compactions by TSDB later.

The `--max-block-duration` flag allows the user to configure a maximum duration of blocks. The backfilling tool will pick a suitable block duration no larger than this.

While larger blocks may improve the performance of backfilling large datasets, drawbacks exist as well. Time-based retention policies must keep the entire block around if even one sample of the (potentially large) block is still within the retention policy. Conversely, size-based retention policies will remove the entire block even if the TSDB only goes over the size limit in a minor way.

Therefore, backfilling with few blocks, thereby choosing a larger block duration, must be done with care and is not recommended for any production instances.
```